src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -73,8 +73,12 @@ export class FetchPlugin extends MonkeyPatched<Window, 'fetch'> {
         this.enable();
     }
 
-    private isTracingEnabled = () => {
-        return this.context.config.enableXRay;
+    private isTracingEnabled = (url: string) => {
+        return (
+            this.context.config.enableXRay &&
+            this.config.urlsToTrace.length &&
+            this.config.urlsToTrace.every((re) => re.test(url))
```

Comment:
thought: we should make this clear in documentation. some customers might think if a url fails any one of the regex tests, it would not be included, which is not the case here.